pages/transactionUpload.tsx

```diff
@@ -0,0 +1,234 @@
+  const [codeEditorContents, setCodeEditorContents] = useState("");
+  const [hasLintError, setHasLintError] = useState(true);
+  const [openResponseModal, setOpenResponseModal] = useState(false);
+  const [modalContents, setModalContents] = useState<fhirJson.BundleEntry[] | null>(null);
+
+  return (
+    <div style={{ paddingLeft: "15px", paddingRight: "15px" }}>
+      <Modal
+        data-testid="transaction-response-modal"
+        opened={openResponseModal}
+        onClose={() => setOpenResponseModal(false)}
+        size="60%"
+        centered
+        overflow="inside"
+        title={
+          <>
+            <Title order={1} align="center" style={{ marginRight: "0px" }} key="modal-title">
+              <Check color="green" size={40} style={{ paddingRight: 8 }} key="green-check" />
+              Transaction Bundle Upload Successful!
+            </Title>
+            <Divider />
+          </>
+        }
+      >
+        <Stack
+          align="flex-start"
+          justify="space-between"
+          style={{ paddingLeft: 20 }}
+          spacing="xl"
+          key="modal-contents"
+        >
+          {processBundleResponse(modalContents)}
+        </Stack>
+      </Modal>
+      <BackButton />
+      <Stack spacing="xs">
+        <Center key="Center-1">
+          <h2 style={{ color: textGray, marginTop: "0px", marginBottom: "8px" }}>
+            Upload Transaction Bundle
+          </h2>
+        </Center>
+        <Center key="Center-2">
+          <Text size="md" color={textGray}>
+            Enter valid transaction bundle body. Then click the Upload button.
+          </Text>
+        </Center>
+        <Center key="Center-3">
+          <Text size="xs" color={textGray}>
+            Note: Only PUT and POST requests are supported in a TransactionBundle at this time.
+          </Text>
+        </Center>
+      </Stack>
+      <div style={{ paddingTop: "20px", paddingBottom: "20px" }}>
+        <ResourceCodeEditor
+          initialValue=""
+          onUpdate={(submittedVal) => setCodeEditorContents(submittedVal)}
+          onValidate={(hasError) => setHasLintError(hasError)}
+        />
+      </div>
+      <Center>
+        <Button
+          data-testid="upload-button"
+          disabled={hasLintError}
+          onClick={editorSubmitHandler}
+          color="cyan"
+          variant="filled"
+          size="lg"
+        >
+          Upload Bundle
+        </Button>
+      </Center>
+    </div>
+  );
+
+  //called when the upload button is clicked. Handles POST request and response
+  async function editorSubmitHandler() {
+    let customMessage = <Text weight={500}>Problem connecting to server:&nbsp;</Text>;
+    let uploadSuccessful = false;
+
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/`, {
+      method: "POST",
+      body: codeEditorContents,
+      headers: {
+        "Content-Type": "application/json+fhir",
+      },
+    })
+      .then((response) => {
+        return response;
+      })
```

Comment:
This .then can be removed, since the next .then will wait for response to be resolved anyway